packages/material-ui/src/InputBase/InputBase.js

```diff
@@ -180,8 +180,9 @@ const InputBase = React.forwardRef(function InputBase(props, ref) {
   } = props;
 
   const inputRef = React.useRef();
-  const { current: isControlled } = React.useRef(props.value != null);
+  const { current: isControlled } = React.useRef(value != null);
   const [focused, setFocused] = React.useState(false);
+  const muiFormControlRef = React.useRef(null);
```

Comment:
You might want to use an initial value here. Otherwise muiFormControlRef.current might point to null for a short period.

Reply:
Should this be the initial value `const muiFormControlRef = React.useRef(muiFormControl);` ?

Reply:
Yeah, Either that or `{}` I think we should add a default value of {} anyway so we can remove some defensive checks.

Reply:
done.